Mathlib/CategoryTheory/Monad/Monadicity.lean

```diff
@@ -0,0 +1,490 @@
+    IsColimit (unitCofork (G := G) A) :=
+  isColimitOfHasCoequalizerOfPreservesColimit G _ _
+#align category_theory.monad.monadicity_internal.unit_colimit_of_preserves_coequalizer CategoryTheory.Monad.MonadicityInternal.unitColimitOfPreservesCoequalizer
+
+/-- The counit cofork is a colimit provided `G` reflects it. -/
+def counitCoequalizerOfReflectsCoequalizer (B : D)
+    [ReflectsColimit (parallelPair ((leftAdjoint G).map (G.map (adj.counit.app B)))
+      (adj.counit.app ((leftAdjoint G).obj (G.obj B)))) G] :
+    IsColimit (counitCofork (G := G) B) :=
+  isColimitOfIsColimitCoforkMap G _ (beckCoequalizer ((comparison adj).obj B))
+#align category_theory.monad.monadicity_internal.counit_coequalizer_of_reflects_coequalizer CategoryTheory.Monad.MonadicityInternal.counitCoequalizerOfReflectsCoequalizer
+
+-- Porting note: Lean 3 didn't seem to need this
+instance [∀ A : (Adjunction.ofRightAdjoint G).toMonad.Algebra,
+    HasCoequalizer ((leftAdjoint G).map A.a) ((Adjunction.ofRightAdjoint G).counit.app
+    ((leftAdjoint G).obj A.A))] (B : D)  : HasColimit (parallelPair ((leftAdjoint G).map
+      (G.map (NatTrans.app (Adjunction.ofRightAdjoint G).counit B)))
+      (NatTrans.app (Adjunction.ofRightAdjoint G).counit ((leftAdjoint G).obj (G.obj B)))) :=
+  inferInstanceAs <| HasCoequalizer
+    ((leftAdjoint G).map ((comparison (Adjunction.ofRightAdjoint G)).obj B).a)
+    ((Adjunction.ofRightAdjoint G).counit.app ((leftAdjoint G).obj
+      ((comparison (Adjunction.ofRightAdjoint G)).obj B).A))
+  --
+theorem comparisonAdjunction_counit_app
+    [∀ A : (Adjunction.ofRightAdjoint G).toMonad.Algebra,
+      HasCoequalizer ((leftAdjoint G).map A.a) ((Adjunction.ofRightAdjoint G).counit.app
+      ((leftAdjoint G).obj A.A))] (B : D) :
+    (comparisonAdjunction (G := G)).counit.app B = colimit.desc _ (counitCofork B) := by
+  apply coequalizer.hom_ext
+  change
+    coequalizer.π _ _ ≫ coequalizer.desc ((adj.homEquiv _ B).symm (𝟙 _)) _ =
+      coequalizer.π _ _ ≫ coequalizer.desc _ _
+  simp
+#align category_theory.monad.monadicity_internal.comparison_adjunction_counit_app CategoryTheory.Monad.MonadicityInternal.comparisonAdjunction_counit_app
+
+end
+
+end MonadicityInternal
+
+open CategoryTheory.Adjunction
+
+open MonadicityInternal
+
+variable {C : Type u₁} {D : Type u₂}
+
+variable [Category.{v₁} C] [Category.{v₁} D]
+
+variable (G : D ⥤ C)
+
+/--
+If `G` is monadic, it creates colimits of `G`-split pairs. This is the "boring" direction of Beck's
+monadicity theorem, the converse is given in `monadic_of_creates_G_split_coequalizers`.
+-/
+def createsGSplitCoequalizersOfMonadic [MonadicRightAdjoint G] ⦃A B⦄ (f g : A ⟶ B)
+    [G.IsSplitPair f g] : CreatesColimit (parallelPair f g) G := by
+  apply (config := {allowSynthFailures := true}) monadicCreatesColimitOfPreservesColimit
+    -- Porting note: apparently cannot do synth failures = true here?
```

Comment:
What happens? Perhaps the porting note could explain.

Reply:
Updated. Behaved same as `:= false`